mcda/utils/utils_for_main.py

```diff
@@ -0,0 +1,1200 @@
+
+    Parameters:
+    - input_matrix_path (str): Path to the CSV file containing the input matrix.
+
+    Raises:
+    - Exception: If an error occurs during the file reading or DataFrame creation.
+
+    :rtype: pd.DataFrame
+    :param input_matrix_path: str
+    """
+
+    try:
+        filename = input_matrix_path
+        with open(filename, 'r') as fp:
+            matrix = pd.read_csv(fp, sep="[,;:]", decimal='.', engine='python')
+            data_types = {col: 'float64' for col in matrix.columns[1:]}
+            matrix = matrix.astype(data_types)
+            alternatives_column_name = matrix.columns[0]
+            matrix = matrix.set_index(alternatives_column_name)
+            return matrix
+    except Exception as e:
+        print(e)
+
+
+def reset_index_if_needed(series):
+    """
+    Reset the index of a pandas Series if it's not a RangeIndex.
+
+    Parameters:
+    - series (pd.Series): The input pandas Series.
+
+    Returns:
+    - pd.Series: The series with the index reset if needed.
+
+    # Usage example:
+    res = reset_index_if_needed(res)
+    """
+    if not isinstance(series.index, pd.RangeIndex):
+        series = series.reset_index(drop=True)
+    return series
+
+
+def _check_and_rescale_negative_indicators(input_matrix: pd.DataFrame) -> pd.DataFrame():
+    """If some indicators in the input matrix are negative, they are rescaled into [0-1]"""
+
+    if (input_matrix < 0).any().any():
+        scaler = MinMaxScaler()
+        scaled_data = scaler.fit_transform(input_matrix)
+        scaled_matrix = pd.DataFrame(
+            scaled_data, columns=input_matrix.columns, index=input_matrix.index)
+        return scaled_matrix
+    else:
+        return input_matrix
+
+
+def parse_args():
+    """
+    Parse command line arguments for configuration path.
+
+    Raises:
+    - argparse.ArgumentError: If the required argument is not provided.
+
+    :rtype: str
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--config', help='config path', required=True)
+    args = parser.parse_args()
+
+    return args.config
+
+
+def get_config(config_path: str) -> dict:
+    """
+    Load and return a configuration dictionary from a JSON file.
+
+    Raises:
+    - Exception: If an error occurs during file reading or JSON decoding.
+
+    :rtype: dict
+    :param config_path: str
+    """
+
+    try:
+        with open(config_path, 'r') as fp:
+            return json.load(fp)
+    except Exception as e:
+        print(e)
+
+
+def save_df(df: pd.DataFrame, folder_path: str, filename: str):
+    """
+    Save a DataFrame to a CSV file with a timestamped filename.
+
+    Parameters:
+    - df (pd.DataFrame): The DataFrame to be saved.
+    - folder_path (str): The path to the folder where the file will be saved.
+    - filename (str): The original filename for the CSV file.
+
+    Notes:
+    - The saved file will have a timestamp added to its filename.
+
+    Example:
```

Comment:
I think it is a good idea to give an example call. But again, I think it should either be in all of the methods or none.